typescript/src/schema/proto.ts

```diff
@@ -99,7 +100,25 @@ function pagingFieldName(messages: MessagesMap, method: MethodDescriptorProto) {
   if (repeatedFields.length !== 1) {
     return undefined;
   }
-  return repeatedFields[0].name;
+  return repeatedFields[0];
+}
+
+function pagingFieldName(messages: MessagesMap, method: MethodDescriptorProto) {
+  const repeatedFields = pagingField(messages, method);
+  if (repeatedFields && repeatedFields.name) return repeatedFields.name;
+  else return undefined;
+}
+
+function pagingResponseType(
+  messages: MessagesMap,
+  method: MethodDescriptorProto
+) {
+  const repeatedFields = pagingField(messages, method);
+  if (repeatedFields && repeatedFields.typeName) {
+    const typeName = repeatedFields.typeName; //.google.showcase.v1beta1.EchoResponse
+    return typeName.replace(/.([^.]*)$/, '.I$1');
```

Comment:
A dot is a special character in regexes and must be written as \. if it's not within a character class. Outside of a character class, . means 'any character'. Your regex works because it's applied from left to right and it turns out that the final dot is the first character that starts the first possible match, but still, let's make it /\.([^.]+)$/ (+ instead of * just in case).